File: trunk/src/main/java/net/esper/util/DebugFacility.java
package net.esper.util;

import antlr.DumpASTVisitor;
import antlr.collections.AST;

/**
 * Utility class for enabling certain debugging using system properties.
 */
public class DebugFacility
{
    private final static String PROPERTY_ENABLED_AST_DUMP = "ENABLE_AST_DUMP";

    /**
     * Dump the AST node to system.out.
     * @param ast to dump
     */
    public static void dumpAST(AST ast)
    {
        if (System.getProperty(PROPERTY_ENABLED_AST_DUMP) != null)
        {
            (new DumpASTVisitor()).visit(ast);
        }
    }
}

File: NEsper/util/DebugFacility.cs
using System;
using System.Configuration;

using antlr;
using antlr.collections;

namespace net.esper.util
{
	/// <summary>
	///  Utility class for enabling certain debugging using system properties.
	/// </summary>
	
	public class DebugFacility
	{
		private const String PROPERTY_ENABLED_AST_DUMP = "ENABLE_AST_DUMP";
		
		/// <summary> Dump the AST node to system.out.</summary>
		/// <param name="ast">to dump
		/// </param>
		
		public static void DumpAST(AST ast)
		{
			if ( ConfigurationManager.AppSettings[PROPERTY_ENABLED_AST_DUMP] != null)
			{
				(new DumpASTVisitor()).visit(ast);
			}
		}
	}
}
